generic_server.cc

```diff
@@ -134,7 +136,7 @@ future<> connection::shutdown()
     return make_ready_future<>();
 }
 
-server::server(const sstring& server_name, logging::logger& logger)
+server::server(const sstring& server_name, logging::logger& logger, const db::config& db_cfg)
```

Comment:
In general it's better for services to have a private config struct and not depend on the global configuration. Esp. for a generic thing like generic_server.

Reply:
added